datacube_apps/stacker/fixer.py

```diff
@@ -0,0 +1,273 @@
+        write_data_variables(data.data_vars, nco)
+        nco.close()
+
+        temp_filename.rename(output_filename)
+
+        if config.get('check_data_identical', False):
+            new_tile = make_updated_tile(unwrapped_datasets, output_uri, tile.geobox)
+            new_data = datacube.api.GridWorkflow.load(new_tile, dask_chunks=chunk_profile)
+            check_identical(data, new_data, output_filename)
+
+    except Exception as e:
+        if temp_filename.exists():
+            temp_filename.unlink()
+        raise e
+
+    return unwrapped_datasets, output_uri
+
+
+def write_data_variables(data_vars, nco):
+    for name, variable in data_vars.items():
+        try:
+            with dask.set_options(get=dask.async.get_sync):
+                da.store(variable.data, nco[name], lock=True)
+        except ValueError:
+            nco[name][:] = netcdf_writer.netcdfy_data(variable.values)
+        nco.sync()
+
+
+def check_identical(data1, data2, output_filename):
+    with dask.set_options(get=dask.async.get_sync):
+        if not all((data1 == data2).all().values()):
+            _LOG.error("Mismatch found for %s, not indexing", output_filename)
+            raise ValueError("Mismatch found for %s, not indexing" % output_filename)
+    return True
+
+
+def make_updated_tile(old_datasets, new_uri, geobox):
+    def update_dataset_location(labels, dataset):
+        # type: (object, Dataset) -> list
+        new_dataset = copy.copy(dataset)
+        new_dataset.uris = [new_uri]
+        return [new_dataset]
+
+    updated_datasets = xr_apply(old_datasets, update_dataset_location, dtype='O')
+    return datacube.api.Tile(sources=updated_datasets, geobox=geobox)
+
+
+def process_result(index, result):
+    datasets, new_uri = result
+    for dataset in datasets.values:
+        _LOG.info('Updating dataset location: %s', dataset.local_path)
+        old_uri = dataset.local_uri
+        index.datasets.add_location(dataset.id, new_uri)
+        index.datasets.archive_location(dataset.id, old_uri)
+
+
+@click.command(name=APP_NAME)
+@datacube.ui.click.pass_index(app_name=APP_NAME)
+@datacube.ui.click.global_cli_options
+@click.option('--cell-index', 'cell_index', help='Limit the process to a particular cell (e.g. 14,-11)',
+              callback=task_app.validate_cell_index, default=None)
+@click.option('--year', 'time', callback=task_app.validate_year, help='Limit the process to a particular year')
+@click.option('--export-path', 'export_path',
+              help='Write the stacked files to an external location without updating the index',
+              default=None,
+              type=click.Path(exists=True, writable=True, file_okay=False))
+@task_app.queue_size_option
+@task_app.task_app_options
+@task_app.task_app(make_config=make_fixer_config, make_tasks=make_fixer_tasks)
+def fixer(index, config, tasks, executor, queue_size, **kwargs):
+    click.echo('Starting fixer utility...')
```

Comment:
Doesn't necessarily need to change, but we should try and be consistent about `_LOG` vs print to stdout. I suspect this should be a log because this won't generally be run interactively?